db/dcrpg/queries.go

```diff
@@ -3315,28 +3316,63 @@ func retrieveWindowStats(ctx context.Context, db *sql.DB, charts *cache.ChartDat
 // and tickets count is done here.
 func appendWindowStats(charts *cache.ChartData, rows *sql.Rows) error {
 	defer closeRows(rows)
+
 	windows := charts.Windows
-	windowSize := uint64(charts.DiffInterval)
-	var ticketsCount uint64
+	windowSize := int(charts.DiffInterval)
+
+	fullWindows := len(windows.TicketPrice)
+	if fullWindows > 0 {
+		// Trim off the last window, as done by charts.MissedVotesTip(), since
+		// it is not known to be complete.
+		windows.TicketPrice = windows.TicketPrice[0 : fullWindows-1]
+		windows.PowDiff = windows.PowDiff[0 : fullWindows-1]
+		windows.Time = windows.Time[0 : fullWindows-1]
+		windows.StakeCount = windows.StakeCount[0 : fullWindows-1]
+		fullWindows = len(windows.TicketPrice)
+	}
+	nextWindowHeight := windowSize * (fullWindows + 1)
+	fmt.Println(fullWindows, nextWindowHeight)
+
+	var price, ticketsCount uint64
+	var timestamp time.Time
+	var difficulty float64
+	var fullWindow bool
 	for rows.Next() {
-		var timestamp time.Time
-		var difficulty float64
-		var price, height, count uint64
+		var height int
+		var count uint64
 		if err := rows.Scan(&price, &timestamp, &difficulty, &height, &count); err != nil {
 			return err
 		}
-
 		ticketsCount += count
-		if (height % windowSize) == 0 {
+
+		// If that was the last block in the current sdiff window, append the
+		// data, and reset for the next window.
+		fullWindow = height == nextWindowHeight-1 // e.g. mainnet block 143, 287, etc.
+		if fullWindow {
 			windows.TicketPrice = append(windows.TicketPrice, price)
 			windows.PowDiff = append(windows.PowDiff, difficulty)
 			windows.Time = append(windows.Time, uint64(timestamp.Unix()))
 			windows.StakeCount = append(windows.StakeCount, ticketsCount)
 
+			// Next sdiff window
 			ticketsCount = 0
-		}
+			nextWindowHeight += windowSize
+		} else if height == nextWindowHeight {
+			return fmt.Errorf("reach height %d before the end of an sdiff window at %d",
+				height, nextWindowHeight)
+		} // else height < nextWindowHeight-1
 	}
 
+	// Store the partial window miss count.
+	if !fullWindow {
+		windows.TicketPrice = append(windows.TicketPrice, price)
+		windows.PowDiff = append(windows.PowDiff, difficulty)
+		windows.Time = append(windows.Time, uint64(timestamp.Unix()))
+		windows.StakeCount = append(windows.StakeCount, ticketsCount)
+	}
```

Comment:
A partial window (the current window) is included now.

Reply:
It's a little confusing that the last data point will not be indicative of a full window. For instance, the ticket count of the last point will only be the count of a single block's tickets at one point. 

Not from this PR, but the comment above this block seems like copypasta.

Reply:
Yeah, that's true.  This change wasn't necessary.  Will revert.

Reply:
I've addressed the review comments.  It works with the test db and a full db.